src/pudl/transform/eia861.py

```diff
@@ -920,6 +919,38 @@ def dynamic_pricing(tfr_dfs):
         dict: A dictionary of transformed EIA 861 dataframes, keyed by table name.
 
     """
+    idx_cols = [
+        "utility_id_eia",
+        "state",
+        "balancing_authority_code_eia",
+        "report_date",
+    ]
+
+    raw_dp = tfr_dfs["dynamic_pricing_eia861"].copy()
+
+    ###########################################################################
+    # Tidy Data:
+    ###########################################################################
+
+    logger.info("Tidying the EIA 861 Dynamic Pricing table.")
+    tidy_dp = _tidy_customer_class_dfs(raw_dp, idx_cols)
+
+    ###########################################################################
+    # Transform Values:
+    # * Make Y/N's into booleans
+    ###########################################################################
+
+    logger.info(
+        "Performing value transformations on EIA 861 Dynamic Pricing table.")
+    transformed_dp = (
+        tidy_dp.replace({'Y': True, 'N': False})
+    )
```

Comment:
This is dangerous because you're telling it that Y/N should be replaced with True/False in *any* column across the whole dataframe, rather than doing the replacement only in the appropriate columns.  If there's only a single column that needs replacing you can do this like:
```py
df["column_name"] = df.column_name.replace({'Y': True, 'N': False})
```
If there's more than one column that needs to be replaced there's a few different ways you can do it. One is using `df.assign` like this:
```py
df = df.assign(
    column_one=lambda x: x.column_one.replace({'Y': True, 'N': False}),
    column_two=lambda x: x.column_two.replace({'Y': True, 'N': False}),
)
```
Which is admittedly kind of inelegant and repetitive.

In some of the EIA 860 transform functions we're doing a bunch of this in a kind of janky way, that's really crying out to be functionalized:
```py
    boolean_columns_to_fix = [
        'duct_burners',
        'multiple_fuels',
        'deliver_power_transgrid',
        'syncronized_transmission_grid',
        'solid_fuel_gasification',
        'pulverized_coal_tech',
        'fluidized_bed_tech',
        'subcritical_tech',
        'supercritical_tech',
        'ultrasupercritical_tech',
        'carbon_capture',
        'stoker_tech',
        'other_combustion_tech',
        'cofire_fuels',
        'switch_oil_gas',
        'bypass_heat_recovery',
        'associated_combined_heat_power',
        'planned_modifications',
        'other_planned_modifications',
        'uprate_derate_during_year',
        'previously_canceled'
    ]

    for column in boolean_columns_to_fix:
        gens_df[column] = (
            gens_df[column]
            .fillna("NaN")
            .replace(
                to_replace=["Y", "N", "NaN"],
                value=[True, False, pd.NA])
        )
```
You probably can't *just* use `True` and `False` because sometimes there won't be any value at all, and the normal `bool` type is not nullable, while the `pandas.BooleanDtype()` is. But you'll have to take the string column that you start with, and swap in `pd.NA` somehow along with the appropriate `True` and `False` replacements.

Reply:
This makes sense - I thought about doing the assign method when writing it, but decided to use replace because it didn't seem to impact the other columns. It's probably good for me to shed this habit!!

Regarding the dtype - does this mean that I'll have to assign the column as object or something similar so that NA values are included?

Reply:
Initially it'll be an `object` column, because that's what pandas uses for strings by default, and that's fine -- it means you can put any value you want in there. With a boolean column your goal is to put only values that are compatible with the `pd.BooleanDtype()` type in there, which means just `True` and `False` and `pd.NA` -- then when the column is *cast* to `pd.BooleanDtype()` by the `cols_dtypes` function, it won't have to guess or complain about making the conversion from `object` to boolean -- all the values will be compatible with the new type.

Reply:
ah I see - so pd.NA is compatible with Boolean?

Reply:
Yep.